File: Source/CSharp/Expressive/Expressive/Functions/AsinFunction.cs
﻿using Expressive.Expressions;
using System;

namespace Expressive.Functions
{
    internal class AsinFunction : FunctionBase
    {
        #region FunctionBase Members

        public override string Name { get { return "Asin"; } }

        public override object Evaluate(IExpression[] parameters)
        {
            this.ValidateParameterCount(parameters, 1, 1);

            return Math.Asin(Convert.ToDouble(parameters[0].Evaluate(Variables)));
        }

        #endregion
    }
}

File: Source/Java/Expressive/src/main/java/com/bijington/expressive/functions/mathematical/AsinFunction.java
package com.bijington.expressive.functions.mathematical;

import com.bijington.expressive.exceptions.ExpressiveException;
import com.bijington.expressive.expressions.IExpression;
import com.bijington.expressive.functions.FunctionBase;

/**
 * Created by shaun on 12/07/2016.
 */
public class AsinFunction extends FunctionBase {
    @Override
    public String getName() {
        return "Asin";
    }

    @Override
    public Object evaluate(IExpression[] parameters) throws ExpressiveException {
        this.validateParameterCount(parameters, 1, 1);

        Object value = parameters[0].evaluate(this.getVariables());

        if (value == null) return null;

        if (value.getClass().equals(Integer.class)) {
            return Math.asin(Integer.class.cast(value));
        }
        else if (value.getClass().equals(Long.class)) {
            return Math.asin(Long.class.cast(value));
        }
        else if (value.getClass().equals(Double.class)) {
            return Math.asin(Double.class.cast(value));
        }
        else if (value.getClass().equals(Float.class)) {
            return Math.asin(Float.class.cast(value));
        }

        return null;
    }
}
